### PR TITLE
Feat/presigned url btn

### PR DESCRIPTION
**Pull request by mfshao**
`Generate Sign URL` button for files, which can be enabled by putting `"signedURLButton": true` in `featureFlags` block of portal config

### New Features
- New `Generate Sign URL` button in file info page, requires `"signedURLButton": true` in `featureFlags` block

### Bug Fixes
- Add check for files that don't have required `data_format` field in metadata and prevent file info page from crashing in that case

### CODE REVIEW

**Comment by mfshao** [comment]
locally tested against `gen3qa-brain`, will figure out which QA env to deploy later

**Comment by mfshao** [comment]
deployed on my dev env

here is a good file that can gives you a signed url: https://mingfei.planx-pla.net/files/1cb65d78-fd84-456a-b5f2-24571293b823
here is a dummy file that can only gives you an error: https://mingfei.planx-pla.net/files/00e6f5f4-0c71-4bc9-bd5e-77a2a494b3b9

File: src/CoreMetadata/CoreMetadataHeader.jsx [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
 import React, { Component } from 'react';
-import {
-  userapiPath,
-} from '../configs';
+import Popup from '../components/Popup';
+import { userapiPath } from '../configs';
 
 const DOWNLOAD_BTN_CAPTION = 'Download';
+const SIGNED_URL_BTN_CAPTION = 'Generate Signed URL';
+const SIGNED_URL_MSG = 'Please copy your signed URL below (this generated signed URL will expires after 3600 seconds):';
```
Comment (qingyashu) [inline src/CoreMetadata/CoreMetadataHeader.jsx]: `expires` -> `expire`

Comment (qingyashu) [reply src/CoreMetadata/CoreMetadataHeader.jsx]: `3600 seconds` -> `1 hour` ? (idk, just more human readable

File: src/CoreMetadata/CoreMetadataHeader.jsx [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
+const SIGNED_URL_ERROR_MSG = 'A error has occurred when generating signed URL:';
```
Comment (qingyashu) [inline src/CoreMetadata/CoreMetadataHeader.jsx]: `A` -> `An`

File: src/CoreMetadata/CoreMetadataHeader.jsx [hunk 3]
```diff
@@ -44,6 +48,15 @@ function canUserDownload(user, projectAvail, projectID) {
 }
 
 class CoreMetadataHeader extends Component {
+  onGenerateSignedURL = () => {
+    this.props.onGenerateSignedURL(this.props.metadata.object_id);
+  };
+
+  onSignedURLPopupClose = () => {
+    this.props.onUpdatePopup({ signedURLPopup: false });
+    this.props.onClearSignedURL();
```
Comment (qingyashu) [inline src/CoreMetadata/CoreMetadataHeader.jsx]: 🤔 I don't understand here why do we need to clear url?

Comment (mfshao) [reply src/CoreMetadata/CoreMetadataHeader.jsx]: I was just thinking to be secure? So if u close the popup then you lose it

but might be an overkill...

Comment (qingyashu) [reply src/CoreMetadata/CoreMetadataHeader.jsx]: i see... maybe we don't need to worry about that :p up to you

**Pull request merged**